src/neural_mesh_simplification/losses/combined_loss.py

```diff
@@ -26,29 +32,40 @@ def __init__(
     def forward(self, original_data, simplified_data):
         original_x = (
             original_data["pos"] if "pos" in original_data else original_data["x"]
-        )
-        original_face = original_data["face"]
-        sampled_indices = simplified_data["sampled_indices"]
-        sampled_vertices = simplified_data["sampled_vertices"]
-        sampled_probs = simplified_data["sampled_probs"]
+        ).to(self.device)
+        original_face = original_data["face"].to(self.device)
+
+        sampled_vertices = simplified_data["sampled_vertices"].to(self.device)
+        sampled_probs = simplified_data["sampled_probs"].to(self.device)
+        sampled_faces = simplified_data["simplified_faces"].to(self.device)
+        face_probs = simplified_data["face_probs"].to(self.device)
 
         chamfer_loss = self.prob_chamfer_loss(
             original_x, sampled_vertices, sampled_probs
         )
+
+        del sampled_probs
+
         surface_loss = self.prob_surface_loss(
             original_x,
             original_face,
             sampled_vertices,
-            simplified_data["simplified_faces"],
-            simplified_data["face_probs"],
+            sampled_faces,
+            face_probs,
         )
         collision_loss = self.collision_loss(
             sampled_vertices,
-            simplified_data["simplified_faces"],
-            simplified_data["face_probs"],
+            sampled_faces,
+            face_probs,
         )
-        edge_crossing_loss = self.edge_crossing_loss(simplified_data)
-        overlapping_triangles_loss = self.overlapping_triangles_loss(simplified_data)
+        edge_crossing_loss = self.edge_crossing_loss(sampled_vertices, sampled_faces, face_probs)
+
+        del face_probs
+
+        overlapping_triangles_loss = self.overlapping_triangles_loss(sampled_vertices, sampled_faces)
+
+        del sampled_vertices
+        del sampled_faces
```

Comment:
Isn't it safe to delete these as well?

```suggestion
        del sampled_vertices
        del sampled_faces
        del original_x
        del original_face

```